flow360/component/geometry.py

```diff
@@ -47,13 +54,132 @@ def _post_upload_heartbeat(info):
         time.sleep(HEARTBEAT_INTERVAL)
 
 
+class GeometryStatus(Enum):
+    """Status of geometry resource, the is_final method is overloaded"""
+
+    COMPLETED = "completed"
+    ERROR = "error"
+    DIVERGED = "diverged"
+    UPLOADED = "uploaded"
+    CASE_UPLOADED = "case_uploaded"
+    UPLOADING = "uploading"
+    RUNNING = "running"
+    PREPROCESSING = "preprocessing"
+    GENERATING = "generating"
+    PROCESSED = "processed"
+    STOPPED = "stopped"
+    DELETED = "deleted"
+    PENDING = "pending"
+    UNKNOWN = "unknown"
+
+    def is_final(self):
+        """
+        [Overloaded] Checks if status is final for geometry resource
+
+        Returns
+        -------
+        bool
+            True if status is final, False otherwise.
+        """
+        if self in [
+            GeometryStatus.COMPLETED,
+            GeometryStatus.DIVERGED,
+            GeometryStatus.ERROR,
+            GeometryStatus.PROCESSED,
+            GeometryStatus.DELETED,
+        ]:
+            return True
+        return False
+
+
 # pylint: disable=R0801
 class GeometryMeta(AssetMetaBaseModel):
     """
     GeometryMeta component
     """
 
     project_id: str = pd.Field(alias="projectId")
+    deleted: bool = pd.Field()
+    metadata: Optional[_GeometryMetadataModel] = pd.Field(None, const=False)
```

Comment:
I think `const=False` is not needed. It is immutable anyway.

Reply:
Right. This was when I tried another way around this.

Reply:
Removed